cmd/infra/azure/destroy.go

```diff
@@ -73,13 +77,29 @@ func (o *DestroyInfraOptions) Run(ctx context.Context, logger logr.Logger) error
 		return fmt.Errorf("failed to create new resource groups client: %w", err)
 	}
 
-	destroyFuture, err = resourceGroupClient.BeginDelete(ctx, o.GetResourceGroupName(), nil)
-	if err != nil {
-		return fmt.Errorf("failed to start deletion: %w", err)
+	var resourceGroups []string
+	resourceGroups = append(resourceGroups, o.ResourceGroupName)
```

Comment:
Could we leverage o.GetResourceGroupName() so hypershift destroy infra can be run without needing to specify --resource-group-name